lib/BC/SleighLifter.cpp

```diff
@@ -687,47 +687,45 @@ class SleighLifter::PcodeToLLVMEmitIntoBlock : public PcodeEmit {
     return LiftStatus::kLiftedUnsupportedInstruction;
   }
 
-  LiftStatus LiftBoolBinOp(llvm::IRBuilder<> &bldr, OpCode opc,
-                           VarnodeData *outvar, VarnodeData lhs,
-                           VarnodeData rhs) {
-    std::function<llvm::Value *(llvm::Value *, llvm::Value *,
-                                llvm::IRBuilder<> &)>
-        op_func;
+
+  std::optional<llvm::Value *>
+  GetResultOfBoolOpFunc(llvm::IRBuilder<> &bldr, OpCode opc,
+                        llvm::Value *in_lhs, llvm::Value *in_rhs) {
     switch (opc) {
       case CPUI_BOOL_AND: {
-        op_func = [](llvm::Value *lhs, llvm::Value *rhs,
-                     llvm::IRBuilder<> &bldr) {
-          return bldr.CreateAnd(lhs, rhs);
-        };
-        break;
+        return bldr.CreateAnd(in_lhs, in_rhs);
       }
       case CPUI_BOOL_OR: {
-        op_func = [](llvm::Value *lhs, llvm::Value *rhs,
-                     llvm::IRBuilder<> &bldr) {
-          return bldr.CreateOr(lhs, rhs);
-        };
-        break;
+
+        return bldr.CreateOr(in_lhs, in_rhs);
       }
       case CPUI_BOOL_XOR: {
-        op_func = [](llvm::Value *lhs, llvm::Value *rhs,
-                     llvm::IRBuilder<> &bldr) {
-          return bldr.CreateXor(lhs, rhs);
-        };
-        break;
+        return bldr.CreateXor(in_lhs, in_rhs);
       }
-      default: break;
+      default: return std::nullopt;
     }
-    if (op_func) {
-      auto lifted_lhs = this->LiftInParam(
-               bldr, lhs, llvm::IntegerType::get(this->context, 8)),
-           lifted_rhs = this->LiftInParam(
-               bldr, rhs, llvm::IntegerType::get(this->context, 8));
-      if (lifted_lhs.has_value() && lifted_rhs.has_value()) {
-        return this->LiftStoreIntoOutParam(
-            bldr, op_func(*lifted_lhs, *lifted_rhs, bldr), outvar);
-      }
+  }
+  LiftStatus LiftBoolBinOp(llvm::IRBuilder<> &bldr, OpCode opc,
+                           VarnodeData *outvar, VarnodeData lhs,
+                           VarnodeData rhs) {
+
+
+    auto lifted_lhs = this->LiftInParam(
+             bldr, lhs, llvm::IntegerType::get(this->context, 8)),
+         lifted_rhs = this->LiftInParam(
+             bldr, rhs, llvm::IntegerType::get(this->context, 8));
+    if (!lifted_lhs.has_value() || !lifted_rhs.has_value()) {
+      return LiftStatus::kLiftedUnsupportedInstruction;
     }
-    return LiftStatus::kLiftedUnsupportedInstruction;
+
+    auto computed_value =
+        this->GetResultOfBoolOpFunc(bldr, opc, *lifted_lhs, *lifted_rhs);
+    if (!computed_value.has_value()) {
```

Comment:
No need to use `has_value()` - this is multiple times in the codebase and while I do not explicitly mind I just don't see the gain of having it there.

Reply:
Can just be `if (!computed_value)`.